src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.Write.cs

```diff
@@ -623,6 +742,24 @@ private static void WriteData(Stream archiveStream, Stream dataStream, long actu
             }
         }
 
+        // Calculates the padding for the current entry and asynchronously writes it after the data.
+        private static ValueTask WritePaddingAsync(Stream archiveStream, long actualLength, CancellationToken cancellationToken)
```

Comment:
The method seems to be just reserving the space for padding by writing zero bytes. I would rename it to make it more obvious

```suggestion
        private static ValueTask WriteEmptyPaddingAsync(Stream archiveStream, long actualLength, CancellationToken cancellationToken)
```